profiling/benchmark.py

```diff
@@ -0,0 +1,360 @@
+# Renamed from examples/llama_example.py
+import argparse
+import contextlib
+import json
+import math
+import os
+import sys
+import time
+from argparse import Namespace
+from typing import Any, Dict, Optional
+
+import torch
+import torch.distributed as dist
+from torch.optim import AdamW
+from torch.profiler import ProfilerActivity
+from tqdm import tqdm
+from transformers import set_seed
+from peft.utils.other import fsdp_auto_wrap_policy
+
+from vectorlm.dataset import Dataset
+from vectorlm.trainer import Trainer
+from vectorlm.utils.data_utils import Config
+from vectorlm.utils.misc_utils import cleanup, setup, wandb_setup
+from vectorlm.utils.model_utils import (
+    hook_activation_checkpointing,
+    load_model_and_tokenizer,
+    shard_model,
+    get_submodule_by_pattern,
+    get_lora_model_from_base_model,
+)
+from vectorlm.utils.optimizer_utils import get_custom_scheduler
+from vectorlm.utils.save_utils import save_consolidated_model
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns
+    -------
+        The parsed arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--yaml_path",
+        default="configs/config.yaml",
+        required=False,
+    )
+    parser.add_argument(
+        "--model_name",
+        required=True,
+    )
+    return parser.parse_args()
+
+
+# unix timestamp
+launch_time = time.time()
```

Comment:
Since `write_metrics` is used strictly within the context of benchmarking, the extra complexity involved in doing so doesn't seem to justify doing so. Please reopen if you think otherwise.